rfcs/0001-multi-tenancy/README.md

```diff
@@ -2,37 +2,82 @@
-This RFC explains two multi-tenancy implementations, their security properties,
-and how they map to the security model.
+To this point, the Flux project has provided [examples of
+multi-tenancy](https://github.com/fluxcd/flux2-multi-tenancy/tree/v0.1.0), but not explained exactly
+how they relate to Flux's authorisation model. This RFC explains two multi-tenancy implementations,
+their security properties, and how they are implemented within the authorisation model.
 
 ### Goals
 
+- Explain the mechanisms available in Flux for supporting multi-tenancy
 - Define two models for multi-tenancy, "soft multi-tenancy" and "hard multi-tenancy".
 - Explain when each model is appropriate.
-- List the tenancy models supported by Flux.
 - Describe a reference implementation of each model with Flux.
 
 ### Non-Goals
 
-- Give an exhaustive account of multi-tenancy implementations.
-- Provide an [end-to-end workflow](](https://github.com/fluxcd/flux2-multi-tenancy/tree/v0.1.0))
-  of how to setup multi-tenancy with Flux.
+- Give an exhaustive account of multi-tenancy implementations in general.
+- Provide an [end-to-end workflow](](https://github.com/fluxcd/flux2-multi-tenancy/tree/v0.1.0)) of
+  how to set up multi-tenancy with Flux.
 
 ## Introduction
 
-Flux allows different organizations and/or teams to share the same Kubernetes control plane.
-Flux enables segmentation and isolation of resources across tenants by using
-Kubernetes Cluster API, namespaces and role-based access control.
+Flux allows different organizations and/or teams to share the same Kubernetes control plane; this is
+referred to as "multi-tenancy". To make this safe, Flux supports segmentation and isolation of
+resources by using namespaces and role-based access control ("RBAC"), and integrating with
+Kubernetes Cluster API.
 
-## User Roles
+The following subsections explain the existing mechanisms used for safe multi-tenancy.
+
+### Flux's authorisation model
+
+Flux defers to Kubernetes' native RBAC to specify which operations are authorised when processing
+the custom resources in the Flux API. By default, this means operations are constrained by the
+service account under which the controllers run, which (again, by default) has the `cluster-admin`
+role bound to it. This is convenient for a deployment in which all users are trusted.
+
+In a multi-tenant deployment, each tenant needs to be restricted in the operations that can be done
+on their behalf. Since tenants control Flux via its API objects, this becomes a matter of attaching
+RBAC rules to Flux API objects. There are two mechanisms that do this, "impersonation" and "remote
+apply".
+
+#### Impersonation
+
+The Kustomize controller and Helm controller both apply arbitrary sets of Kubernetes configuration
+to a cluster. These controllers are subject to authorisation on two counts:
+
+ - when accessing the Kubernetes resources that specify a particular "apply" operation -- for
```

Comment:
This is not what Flux currently does, the sources are not subject to RBAC but to ACL (when RFC-0002 gets implemented).

Reply:
Owps, getting ahead of myself.